### DISCUSSION TITLE
wcl386 -za99 with -aa conundrum

### DISCUSSION THREAD

**Discussion by FredEckert** [open-watcom/open-watcom-v2#727] [Q&A] [answered]
Hi, I am trying to use wcl386 with the -za99 and -aa compiler switches on some C99 code:

-aa : enable non-constant initializers for local aggregates or unions (#33, #34)
[-za99 : enable partial C99 compliance ](https://web.archive.org/web/20150503232909/http://openwatcom.org/index.php/C99_Compliance)

Example test.c program showing problem:
```
#include <stdio.h>
//Open Watcom C/C++ x86 32-bit Compile and Link Utility Version 2.0 beta Aug  1 2021 01:13:02 (64-bit)
//compile: wcl386 -d2 -w3 -za99 test.c          fails with E1054 on line 19
//compile: wcl386 -aa -d2 -w3 -za99 test.c      fails with E1063 on line 14

struct color_t{
        unsigned char red;
        unsigned char green;
        unsigned char blue;
};

int main(void) {
        //test.c(14): Error! E1063: Missing operand (-aa switch breaks this)
        const struct color_t C = { .red = 0x96, .green = 0x4B };
        printf("C.red=%x\nC.green=%x\nC.blue=%x\n", C.red, C.green, C.blue);

        //test.c(19): Error! E1054: Expression must be constant (-aa switch makes this work)
        unsigned char line = 1;
        const unsigned char position[] = {0xFE, 'H', 0xFE, 'G', 0, line};
        printf("%c\n", line);
        printf("%s\n", position);

        return 0;
}
```
I was hoping someone could explain the best way to resolve this situation.

Thank you,
Fred Eckert

**Comment by jmalak** [comment]
Thanks for bug report.
I confirm it is a compiler bug #728

**Reply by FredEckert** [reply]
Thanks for confirming that this is a bug. I wasn't sure. I will not use -aa until the bug is repaired.

**Comment by jmalak** [comment, marked as answer]
Untill it will be fixed you can use -aa option but only for code without designated initializers.

**Reply by jmalak** [reply]
It is fixed in git repository by 24b3f8621880ea12e04199b9a27f0939462089c5 commit.
It will be available in next "Current build", it will run after midnight.

**Reply by FredEckert** [reply]
I checked this fix with current build "Last updated 2021-09-13 01:42:15". The fix worked for the above test.c program using both the windows x64 and the DOS wcl386 builds. I was also able to successfully compile and run a large c99 project with the fix.

**Comment by FredEckert** [comment]
Unrelated additional comment: I have noticed that the help files are not being installed with the Current build.  When installing, I selected "Win32 Hosted Help Files", "Win32 Hosted HTML Help Files", and "PDF Help FIles". After the installation, there were no .chm, .pdf, or .htm files in the C:\WATCOM folder or any subfolder.

**Reply by jmalak** [reply]
Thanks for next bug report.
Installer problem with documentation is fixed by git commit b58710a08dc609fc572efdc07dba27f258d34bd4
new correct installers will be available tomorrow morning after midnight re-build.

**Reply by FredEckert** [reply]
Thank you for the support. I confirm that the current build "Last updated 2021-09-14 02:24:04" fixes the help file install issue on the win-x64 installer.

**Reply by jmalak** [reply]
Thanks for your checking.
